src/workerd/api/http.c++

```diff
@@ -1595,6 +1595,38 @@ jsg::Promise<jsg::Ref<Response>> handleHttpRedirectResponse(jsg::Lock& js,
     return js.rejectedPromise<jsg::Ref<Response>>(kj::mv(exception));
   }
 
+
+  if (FeatureFlags::get(js).getStripAuthorizationOnCrossOriginRedirect()) {
+    auto base = urlList.back().toString();
+    auto currentUrl = jsg::Url::tryParse(base.asPtr());
+
+    if (currentUrl == kj::none) {
+      auto exception =
+        JSG_KJ_EXCEPTION(FAILED, TypeError, "Invalid current URL; unable to follow redirect.");
+      return js.rejectedPromise<jsg::Ref<Response>>(kj::mv(exception));
+    }
+
+
+    auto locationUrl = jsg::Url::tryParse(location, base.asPtr());
+
+    if (locationUrl == kj::none) {
+      auto exception =
+        JSG_KJ_EXCEPTION(FAILED, TypeError, "Invalid Location header; unable to follow redirect.");
+      return js.rejectedPromise<jsg::Ref<Response>>(kj::mv(exception));
+    }
+
+    if (KJ_ASSERT_NONNULL(currentUrl).getOrigin() != KJ_ASSERT_NONNULL(locationUrl).getOrigin()) {
```

Comment:
The KJ_ASSERT_NONNULLs here are redundant since the code above ensures that these cannot possibly be null. This makes the code hard to read: It _looks like_ we're doing checks here that could fail and throw exceptions, and one has to carefully read the code that comes before it to see that's not the case.

You can avoid this problem by changing the checks above to use KJ_UNWRAP_OR, like:

```cpp
    auto& someCurrentUrl = KJ_UNWRAP_OR(currentUrl, {
      auto exception =
        JSG_KJ_EXCEPTION(FAILED, TypeError, "Invalid current URL; unable to follow redirect.");
      return js.rejectedPromise<jsg::Ref<Response>>(kj::mv(exception));
    });
```

Now you can use `someCurrentUrl` without checking it again.